src/edu/mills/cs180a/RomanNumeral.java

```diff
@@ -14,154 +14,175 @@
-			//if that's the case then throw an error
-    		if (LETTERS_TO_VALUES.get(current_letter)==null)
-        	{
-    			throw new IllegalArgumentException(
-   				 current_letter + "is not a valid roman numeral");
-        	}//end if
-    		
-    		//Letters_to_value.get(s[i]).getValue()
-		
-			//compare the current letter to the next letter 
-			//if the current letter is smaller than the next letter, you subtract that value from the total
-			//if the current letter is larger than the next letter, you add that value to the total
-			if (LETTERS_TO_VALUES.get(current_letter)< LETTERS_TO_VALUES.get(next_letter))
-				total -= LETTERS_TO_VALUES.get(current_letter);
-			else
-				total += LETTERS_TO_VALUES.get(current_letter);
-		
-    	}//end for loop
-
-    	return total;
-    }//end convertToInteger
-    
+    protected static int convertFromString(String s) {
+        // allow for lower case Roman Numerals to be accepted
+        char[] letters = s.toUpperCase().toCharArray();
+
+        // throw an error if any of the chars do not exist in the map
+        for (int i = 0; i < letters.length; i++) {
+            if (!LETTERS_TO_VALUES.containsKey(String.valueOf(letters[i]))) {
+                throw new IllegalArgumentException(letters[i] + "is not a valid roman numeral");
+            }
+        }
+
+        int numericValue = LETTERS_TO_VALUES.get(String.valueOf(letters[0]));
+
+        // set the limit to one before the full length of the char array so that we don't run into
+        // any issues with variable next
+        for (int i = 0; i < letters.length - 1; i++) {
+            String current = String.valueOf(letters[i]);
+            String next = String.valueOf(letters[i + 1]);
+
+            if (LETTERS_TO_VALUES.get(current) >= LETTERS_TO_VALUES.get(next)) {
+                numericValue += LETTERS_TO_VALUES.get(next);
+            } else {
+                // Check if current starts a subtractive form (e.g., IX), where the second symbol
+                // has a higher value than the first. Subtract current twice: once because it is the
+                // start of a subtractive form and once because it was added in a previous
+                // iteration.
+                numericValue += LETTERS_TO_VALUES.get(next) - (2 * LETTERS_TO_VALUES.get(current));
+            }
+        }
+        return numericValue;
+    }// end convertFromString
+
+
     /**
-     * Returns the Roman Numeral representation of the given number.
+     * Returns the Roman Numeral representation of any given number by first determining which keys
+     * in {@link #VALUES_TO_LETTERS} the number falls between and setting the lesser key as a
+     * {@link #lowerBound}. The number, {@link #lowerBound} and a StringBuilder are passed to a
+     * helper method, which will append the appropriate Roman symbol to the StringBuilder and return
+     * the number, of which the value has decreased based on the numeric value of the appended
+     * symbol.
```

Comment:
This provides implementation details, which is not appropriate for javadoc. See me if you don't understand this. Change anywhere else you do this. The original javadoc was fine.